src/Diagnostics.HealthChecks/Internal/NonCapturingTimer.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+//Imported from https://github.com/dotnet/aspnetcore/blob/main/src/Shared/NonCapturingTimer/NonCapturingTimer.cs
+
+#nullable enable
+
+using System;
+using System.Threading;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks;
+
+// A convenience API for interacting with System.Threading.Timer in a way
```

Comment:
use summary?

```suggestion
/// <summary>
// A convenience API for interacting with System.Threading.Timer in a way
// that doesn't capture the ExecutionContext. We should be using this (or equivalent)
// everywhere we use timers to avoid rooting any values stored in asynclocals.
/// </summary>
```